src/imperator/data/setup_main.rs

```diff
@@ -0,0 +1,295 @@
+use crate::block::Block;
+use crate::context::ScopeContext;
+use crate::db::{Db, DbKind};
+use crate::effect::validate_effect;
+use crate::everything::Everything;
+use crate::game::GameFlags;
+use crate::item::{Item, ItemLoader};
+use crate::scopes::Scopes;
+use crate::token::Token;
+use crate::tooltipped::Tooltipped;
+use crate::trigger::validate_trigger;
+use crate::validate::validate_modifiers_with_base;
+use crate::validator::Validator;
+use crate::modif::{validate_modifs, ModifKinds};
+use crate::imperator::tables::misc::{DLC_IMPERATOR};
+
+#[derive(Clone, Debug)]
+pub struct SetupMain {}
+
+impl SetupMain {
+    pub fn add(db: &mut Db, key: Token, block: Block) {
+        db.add(Item::Decision, key, block, Box::new(Self {}));
+    }
+}
+
+impl DbKind for SetupMain {
+    fn validate(&self, key: &Token, block: &Block, data: &Everything) {
+        let mut vd = Validator::new(block, data);
+
+        vd.field_validated_block("treasure_manager", |block, data| {
+            validate_treasures(block, data);
+        });
+        vd.field_validated_block("family", |block, data| {
+            validate_families(block, data);
+        });
+        vd.field_validated_block("diplomacy", |block, data| {
+            validate_diplomacy(block, data);
+        });
+        vd.field_validated_block("provinces", |block, data| {
+            validate_provinces(block, data);
+        });
+        vd.field_validated_block("road_network", |block, data| {
+            validate_roads(block, data);
+        });
+        vd.field_validated_block("country", |block, data| {
+            validate_countries(block, data);
+        });
+        vd.field_validated_block("trade", |block, data| {
+            validate_trade(block, data);
+        });
+        vd.field_validated_block("great_work_manager", |block, data| {
+            validate_great_works(block, data);
+        });
+    }
+}
+
+impl FileHandler<Block> for SetupMain {
+    // TODO - Is this impl needed? Copied it from src/vic3/data/history.rs
```

Comment:
This depends om whether SetupMain follows the key = { ... } format of most game item types. If it does, you can use the DbKind validate approach. If it's too different, you have to use the FileHandler approach which is more manual (and is an older API)

Reply:
I think the DbKind validate approach will be enough here in that case, will remove the FileHandler impl then